### PR TITLE
feat: Resolve system properties/environment variables while browsing the application.properties values

### LINKED ISSUE #448: Resolve system properties/environment variables while browsing the application.properties values
Originally reported as https://github.com/redhat-developer/intellij-quarkus/issues/1331

> It would be nice if the current System properties (eg. `user.name`) is resolved while browsing the `application.properties` values:
> image

 it is supported in MP Config according to @gastaldi

`*_KEY`, `*_SECRET` and `*_PASSWORD` should be obfuscated. 

I initially started looking at an IJ implem in https://github.com/fbricon/intellij-quarkus/commit/a22b996b8a32bca298d636930f1d58e8c7a7a1c2 (validation and completion still didn't work), but this should better be done at the language server level directly.

### PR DESCRIPTION
**Pull request by angelozerr**
feat: Resolve system properties/environment variables while browsing the application.properties values

Fixes #448

Here the result:

image

//cc @gastaldi

### CODE REVIEW

File: microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/extensions/sysenv/SysEnvItemMetadataProvider.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		});
+
+		return Stream.concat(sysProps, envVars)//
+				.collect(Collectors.toList());
+
+	}
+
+	private static String obfuscate(String key, String value) {
+		String upKey = key.toUpperCase().replace(".", "_");
+		return upKey.endsWith("_KEY") || upKey.endsWith("_SECRET") ? "*********" : value;
```
Comment (fbricon) [inline microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/extensions/sysenv/SysEnvItemMetadataProvider.java:74]: can you add _PASSWORD too?

Comment (gastaldi) [reply microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/extensions/sysenv/SysEnvItemMetadataProvider.java:74]: Maybe `_TOKEN` as well?

Comment (angelozerr) [reply microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/extensions/sysenv/SysEnvItemMetadataProvider.java]: Added

**Comment by datho7561** [comment]
Hovering on the system property/environment variable names causes an error popup

**Comment by angelozerr** [comment]
@datho7561 I cannot reproduce it, could you share a simple example please.

**Comment by datho7561** [comment]
> I cannot reproduce it, could you share a simple example please.

https://github.com/user-attachments/assets/9fd454af-f08a-46f7-a180-5b9d3af46e4f

The value of `ASDF_DIR` is `/home/davthomp/.asdf`

**Comment by angelozerr** [comment]
> Hovering on the system property/environment variable names causes an error popup, and the following is logged in the language server logs:

> Hovering on the system property/environment variable names causes an error popup, and the following is logged in the language server logs:
> 
> Request failed: Request microprofile/propertyDocumentation failed with message: Command 'microprofile/propertyDocumentation' must be called with required MicroProfilePropertyDocumentationParams.sourceType! (-32603).

It should be fixed.

File: microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/extensions/sysenv/SysEnvItemMetadataProvider.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+				.collect(Collectors.toList());
+
+	}
+
+	private static String obfuscate(String key, String value) {
+		if (StringUtils.isEmpty(key)) {
+			return key;
+		}
+		String upKey = key.toUpperCase().replace(".", "_");
+		return upKey.endsWith("_KEY") || upKey.endsWith("_SECRET") || upKey.endsWith("_PASSOWORD")
```
Comment (datho7561) [inline microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/extensions/sysenv/SysEnvItemMetadataProvider.java:76]: `_PASSOWORD` --> `_PASSWORD`

**Review by datho7561 — CHANGES_REQUESTED** [review]

**Comment by datho7561** [comment]
Sounds good. I checked out the latest commit and confirmed that's working. Once the tests are in place I think it's ready to go.

**Comment by angelozerr** [comment]
> Sounds good. I checked out the latest commit and confirmed that's working. Once the tests are in place I think it's ready to go.

Thanks @datho7561 ! I wrote a lot of tests and I merge it since CI build is working well.

**Pull request merged**